kong/plugins/response-transformer/body_transformer.lua

```diff
@@ -82,39 +101,58 @@ function _M.transform_json_body(conf, buffered_data)
   end
 
   -- replace key:value to body
-  for _, name, value in iter(conf.replace.json) do
+  for i, name, value in iter(conf.replace.json) do
     local v = cjson.encode(value)
     if v and sub(v, 1, 1) == [["]] and sub(v, -1, -1) == [["]] then
       v = gsub(sub(v, 2, -2), [[\"]], [["]]) -- To prevent having double encoded quotes
     end
 
     v = v and gsub(v, [[\/]], [[/]]) -- To prevent having double encoded slashes
+
+    if conf.replace.json_types then
+      local v_type = conf.replace.json_types[i]
+      v = cast_value(v, v_type)
+    end
+
     if json_body[name] and v then
```

Comment:
When `cast_value` converts `v` to boolean, this will be `false` and will not enter this branch. The fix is the following:

```suggestion
    if json_body[name] and v ~= nil then
```

(Note that this implicitly also means that when a conversion fails, the value stays unchanged (and sent as a string)).

Perhaps we need to add test cases for `true` and `false` values as well? It would have caught this.

Reply:
Thanks for the catch, I have made some changes as suggested and added some tests [in this commit](https://github.com/Kong/kong/pull/4889/commits/cad6b59c3824645c3d899f97ff5e145f6e9f15e8) to cover that.